source/ed/osm2ed.h

```diff
@@ -200,14 +215,13 @@ struct OSMWay {
         return bg::distance(p, ls);
     }
 
-    std::set<const OSMRelation*> admins() const {
-        std::set<const OSMRelation*> result;
+    bool contains_admin(const Admin* admin) const {
         for (auto node : nodes) {
-            if (node->admin != nullptr) {
-                result.insert(node->admin);
+            if (node->admin == admin) {
```

Comment:
Or in a less baroque style  :stuck_out_tongue: 

```cpp
bool contains_admin(const Admin* admin) const { 
    auto has_admin = [&](const auto& n){ return n->admin == admin; };
    return boost::find_if(nodes, has_admin) != nodes.end();
}
```

Then, you realise how much boilerplate you need to for a c++ lambda  and you wish to write them the Rust way :sob: :sob: :sob:

Reply:
you're right, it's way better (even if too noisy 😜 ), I'll change this on monday